src/buildVariables.js

```diff
@@ -112,7 +112,16 @@ const buildUpdateVariables = (
     queryType
 ) =>
     Object.keys(params.data).reduce((acc, key) => {
-
+        // If hasura permissions do not allow a field to be updated like (id),
+        // we are not allowed to put it inside the variables
+        // RA passes the whole previous Object here
+        // https://github.com/marmelab/react-admin/issues/2414#issuecomment-428945402
+
+        // TODO: To overcome this permission issue,
+        // it would be better to allow only permitted inputFields from *_set_input INPUT_OBJECT
+        if (params.data[key] === params.previousData[key]) {
+            return acc;
+        }
```

Comment:
Here we are just not sending the untouched field, so if you do not provide an EditField, it should be ok (at least for primitives)